### PR TITLE
Merge 3.0 -> master

### PR DESCRIPTION
**Pull request by jtpio**
There is now a `2.x` branch that points to the old master: https://github.com/jupyterlab/extension-examples/tree/2.x

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -176,8 +182,9 @@ jobs:
       - name: Install the Python dependencies
         if: steps.filter.outputs.extension == 'true'
         run: |
+          python -m pip install --upgrade pip
           python -m pip install jupyter_packaging
-          python -m pip install jupyterlab
+          python -m pip install --pre jupyterlab
```
Comment (fcollonval) [inline .github/workflows/main.yml:187]: Do we want to keep using the `pre` release?

```suggestion
          python -m pip install jupyterlab
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -267,12 +272,11 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-pip-
       - name: Install the Python dependencies
-        run: python -m pip install jupyterlab
+        run: python -m pip install --pre jupyterlab
```
Comment (fcollonval) [inline .github/workflows/main.yml:275]:
```suggestion
        run: python -m pip install jupyterlab
```

**Review by fcollonval — COMMENTED** [review]
Thanks @jtpio!

A small comment: do we want to keep installing the prerelease of JupyterLab?

File: basics/hello-world/README.md [hunk 3]
```diff
@@ -91,11 +91,11 @@ called `@jupyterlab/application`. The dependency of your extension on this
 package is declared in the file `package.json`:
 
 ```json5
-// package.json#L36-L38
+// package.json#L44-L46
```
Comment (fcollonval) [inline basics/hello-world/README.md:94]: Correct readme

```suggestion
// package.json#L46-L48
```

File: .github/workflows/main.yml (hunk 1 above)
Comment (jtpio) [reply .github/workflows/main.yml:187]: Good catch, thanks!

File: advanced/server-extension/README.md [hunk 4]
```diff
@@ -668,19 +689,19 @@ file:
...
 
-"server": {
+"discovery": {
 ```
 
 And that server extension is available through `pip`:
 
 ```json5
-// package.json#L66-L68
+// package.json#L72-L74
```
Comment (fcollonval) [inline advanced/server-extension/README.md:700]: Another one that slide a bit further in the file

```suggestion
// package.json#L73-L75
```

File: main-menu/README.md [hunk 5]
```diff
@@ -85,13 +85,13 @@ After the execution of that command, `package.json` should list them in the
 `dependencies`:
 
 ```json5
-// package.json#L36-L40
+// package.json#L44-L48
```
Comment (fcollonval) [inline main-menu/README.md:88]:
```suggestion
// package.json#L46-L50
```

File: settings/README.md [hunk 6]
```diff
@@ -109,12 +109,11 @@ the `package.json` file in the `jupyterlab` section (here `schema`):
 
 <!-- prettier-ignore-start -->
 ```json5
-// package.json#L57-L60
+// package.json#L68-L70
```
Comment (fcollonval) [inline settings/README.md:112]:
```suggestion
// package.json#L70-L74
```

**Comment by fcollonval** [comment]
@jtpio sorry to prompt them one at a time.

I went through all READMEs to check that the code snippets matches the previous one. Sorry that it will force you to rerun locally embedme.

**Comment by jtpio** [comment]
No problem, thanks for checking this :+1: 

Just pushed the embedme changes.

**Comment by jtpio** [comment]
Looks good now (with one random CI fail).

**Review by fcollonval — APPROVED** [review]
LGTM

**Comment by jtpio** [comment]
All green after re-running the tests, merging.

**Pull request merged**